soularr.py

```diff
@@ -240,99 +243,236 @@ def verify_filetype(file,allowed_filetype):
 
-                        folder_data = {
-                            "artist_name": artist_name,
-                            "release": release,
-                            "dir": file_dir.split("\\")[-1],
-                            "discnumber": track['mediumNumber'],
-                            "username": username,
-                            "directory": directory,
-                        }
-                        grab_list.append(folder_data)
+def download_album(allowed_filetype, username, file_dir, directory, search, retry_list, grab_list, artist_name, release, track):
 
-                        try:
-                            slskd.transfers.enqueue(username = username, files = directory['files'])
-                            # Delete the search from SLSKD DB
-                            if delete_searches:
-                                slskd.searches.delete(search['id'])
-                            logger.info(f"Adding {username} to retry list")
-                            retry_list[username] = {}
-                            for file in directory['files']:
-                                logger.info(f"Adding {file['filename']} to retry list")
-                                retry_list[username][file['filename']] = 0
-                            return True
-                        except Exception as e:
-                            logger.warning(f"Exception {e}")
-                            logger.warning(f"Error enqueueing tracks! Adding {username} to ignored users list.")
-                            downloads = slskd.transfers.get_downloads(username)
+    if download_filtering: 
+        logger.info(f"Processing Download Filtering")
+        if use_extension_whitelist:
+            logger.info(f"Using extensions_whitelist: {use_extension_whitelist}")
+            whitelist = copy.deepcopy(extensions_whitelist)
+        else:
+            whitelist = []
+        whitelist.append(allowed_filetype.split(" ")[0])
+        unwanted = []
+        logger.info(f"Accepted extensions: {whitelist}")
+        for file in directory['files']:
+            match = False
+            for extension in whitelist:
+                if file['filename'].split(".")[-1].lower() == extension.lower():
+                    match = True
+            if not match:
+                unwanted.append(file['filename'])
+                logger.info(f"File: {file['filename']} is unwanted")
+        if len(unwanted) > 0:
+            temp = []
+            for file in directory['files']:
+                if file['filename'] not in unwanted:
+                    logger.info(f"File: {file['filename']} added to modified directory listing")
+                    temp.append(file)
+            directory['files'] = temp
+
+
+    for i in range(0,len(directory['files'])):
+       directory['files'][i]['filename'] = file_dir + "\\" + directory['files'][i]['filename']
```

Comment:
Consider using
```suggestion
    for i in range(0,len(directory['files'])):
       directory['files'][i]['filename'] = os.path.join(file_dir, directory['files'][i]['filename'])
```

And then instead of things like `file_dir.spli("\\")[-1]` use `os.path.basename(file_dir)`.

Reply:
Also maybe consider doing
```suggestion
    directory['files] = [os.path.join(file_dir, file['filename']) for file in directory['files'])]
```

Reply:
That was from the original code before my changes. I haven't looked at it too hard. 

I believe it's done that way so you don't get windows format slashes where you don't want them as that's for building the slskd path. Not the filesystem path .